awd/database.py

```diff
@@ -0,0 +1,106 @@
+from __future__ import annotations
+
+import datetime
+import logging
+from typing import Any
+
+from pynamodb.attributes import NumberAttribute, UnicodeAttribute
+from pynamodb.models import Model
+
+from awd.config import DATE_FORMAT
+from awd.status import Status
+
+logger = logging.getLogger(__name__)
+
+
+class DoiProcessAttempt(Model):
+    """A class modeling the DynamoDB table."""
+
+    class Meta:  # noqa: D106
+        table_name = "None"
+
+    doi = UnicodeAttribute(hash_key=True)
+    attempts = NumberAttribute()
+    last_modified = UnicodeAttribute()
+    status = NumberAttribute()
+
+    def add_item(self, doi: str) -> dict[str, Any]:
+        """Add DOI item to DOI table.
+
+        Args:
+            doi: The DOI to be added to the DOI table.
+        """
+        response = DoiProcessAttempt(
+            doi=doi,
+            attempts=0,
+            last_modified=datetime.datetime.now(tz=datetime.UTC).strftime(DATE_FORMAT),
+            status=Status.UNPROCESSED.value,
+        ).save()
+        logger.debug("%s added to table", doi)
+        return response
```

Comment:
I would suggest a `classmethod` here, as it feels like a factory pattern where a class can return instances of itself:

```python
@classmethod
def add_item(cls, doi: str) -> dict[str, Any]:
    """Add DOI item to DOI table.
    Args:
        doi: The DOI to be added to the DOI table.
    """
    response = cls(
        doi=doi,
        ...
        ...
```